includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding\Services;
+
+use NewfoldLabs\WP\Module\Onboarding\Data\Options;
+use NewfoldLabs\WP\Module\Onboarding\Data\Flows;
+use NewfoldLabs\WP\Module\Onboarding\Data\Data;
+
+class FlowService {
+
+	public static function initalize_flow_data() { 
+        if ( ! ( $flow_data = self::read_flow_data_from_wp_option() ) ) {
+            $flow_data =  self::get_default_flow_data();
+			return self::update_wp_options_data_in_database( $flow_data );
+		}
+		$default_flow_data = Flows::get_data();
+		$updated_flow_data = array();
+		$updated_flow_data = self::update_flow_data_recursive($default_flow_data, $flow_data, $updated_flow_data);
+		return self::update_wp_options_data_in_database($updated_flow_data);
+    }
+
+    public static function get_default_flow_data() {
+		// check if data is available in the database if not then fetch the default dataset
+		$flow_data              = Flows::get_data();
+		$flow_data['createdAt'] = time();
+		// get current flow type
+		$flow_type = Data::current_flow();
+		// update default data if flow type is ecommerce
+		if($flow_type == 'ecommerce') 
+			$flow_data = self::update_default_data_for_ecommerce( $flow_data );
+		self::update_wp_options_data_in_database( $flow_data );
+		return $flow_data;
+	}
+
+	/*
+	 * function to update the Flow Data (Blueprint) in an array recursively in comparison to Flows::get_data() (Database)
+	 */	
+	private static function update_flow_data_recursive(&$default_flow_data, &$flow_data, &$updated_flow_data) {
+		{
```

Comment:
```suggestion
	private static function update_flow_data_recursive(&$default_flow_data, &$flow_data, &$updated_flow_data) {
```
Double opening braces? 👀